comma/model.py

```diff
@@ -16,12 +20,90 @@ def setup(self, size: int, dir_params: str):
 
         Args:
             size (int): size of the population
-
-            dir_params (str): dir to the folder containing feature parameter files.
+            dir_params (str): dir to the folder containing hypothesis and parameter files.
         """
+        self.param_file_validation(dir_params)
         self.agents = Individual.populate(size, dir_params)
         return self.agents
-
+    
+    def param_file_validation(self, dir_params: str):
+        """Validate files in the parameter folder.
+        
+        Args:
+            dir_params (str): dir to the folder containing hypothesis and parameter files.
+        """
+        # check if parameter files exist
+        fpath_params_individual = os.path.join(dir_params, PARAMS_INDIVIDUAL)
+        fpath_params_model = os.path.join(dir_params, PARAMS_MODEL)
+        assert os.path.isfile(fpath_params_individual), \
+            "Parameter file not found: %s." % PARAMS_INDIVIDUAL
+        assert os.path.isfile(fpath_params_model), \
+            "Parameter file not found: %s" % PARAMS_MODEL
+       
+        # check if all required model parameters are given 
+        params_model = read_json_as_dict(fpath_params_model)
+        missing_model_params = set(self._required_params) - set(params_model.keys())
+        assert not missing_model_params, \
+            "Model parameter(s) not found: %s." % ", ".join(missing_model_params)
+            
+        # check if there is any overlap between lockdown intervals
+        lockdown = params_model["lockdown"]
+        lockdown_intervals = [l["interval"] for l in lockdown] 
+        lockdown_intervals = sorted(lockdown_intervals, key=lambda li: li[0])
+        overlaps = [lockdown_intervals[i][1] >= lockdown_intervals[i+1][0] \
+            for i in range(len(lockdown_intervals) - 1)]
+        assert not any(overlaps), "Lockdown intervals have overlap: \n%s" % "\n".join(
+            ["%s - %s" % (str(lockdown_intervals[i]), str(lockdown_intervals[i+1])) \
+                for i in range(len(overlaps)) if overlaps[i]] 
+        )
+        
+        # check if all the steps are fully covered by the given lockdown intervals
+        steps = params_model["steps"]
+        expected_timeline = list(range(1, steps+1))
+        given_timeline = []
+        for li in lockdown_intervals:
+            given_timeline += list(range(li[0], li[1] + 1))
+        uncovered_steps = set(expected_timeline) - set(given_timeline)
+        assert not uncovered_steps, "Uncovered steps: %s." % ", ".join(
+            [str(us) for us in uncovered_steps]
+        )
+            
+        # check if all hypothesis files exist
+        fnames = ["actions_effects_on_%s.csv" % s for s in params_model["status"]]
```

Comment:
The files in the `parameters_example` are named as `action_` not `actions_`. This breaks the code if one runs the model with the example parameters.

However, your function that creates the files automatically it does create them by calling them `actions_` so I suggest renaming the files in the `parameters_example` with `actions_` so that you don't have to change the code unnecessarily.

Reply:
Fixed this.